components/forms/TransactionForm.js

```diff
@@ -26,7 +26,7 @@ class TransactionForm extends React.Component {
 
     this.state = {
       toAddress: "",
-      amount: 0,
+      amount: "0",
```

Comment:
With this change, `state.amount` is a string consistently. Before it was initialized as number and was replaced by strings from the field.